### PR TITLE
UI: Add "18 Scenes" multiview option

### PR DESCRIPTION
**Pull request by XDelta**
### Description
Adds an option for 18 scenes in the multiview. 
This option leaves the preview and program sized to the upper half of the window.

image

### Motivation and Context
Multiview now can show more than the 8 scenes while also not reducing the size of the preview and program windows. 24 scenes visible was already overkill but also has relatively smaller preview and program.

### How Has This Been Tested?
Tested on windows 10 64bit
I tested:
- Each multiview scene was still selecting the correct scene when clicked
- All 18 scenes showed up in multiview.
- Other multiview options were unaffected

The only change that knowingly modifies something outside this specific change is moving `HORIZONTAL_TOP_24_SCENES = 4` down to `HORIZONTAL_TOP_24_SCENES = 5,` to keep the options in order.

### Types of changes
- New feature (non-breaking change which adds functionality)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by XDelta** [comment]
I would suggest changing the 24 scene view translation name from ".Extended.Top" to ".24Scene.Top" or another naming scheme. I didn't want to change the existing translation key with this feature.

**Comment by eazuka** [comment]
Real nice. awesome and thanks for the efforts. I hope this is accepted and merged.

File: UI/window-projector.hpp [hunk 1]
```diff
@@ -18,7 +18,8 @@ enum class MultiviewLayout : uint8_t {
 	HORIZONTAL_BOTTOM_8_SCENES = 1,
 	VERTICAL_LEFT_8_SCENES = 2,
 	VERTICAL_RIGHT_8_SCENES = 3,
-	HORIZONTAL_TOP_24_SCENES = 4,
+	HORIZONTAL_TOP_18_SCENES = 4,
+	HORIZONTAL_TOP_24_SCENES = 5,
```
Comment (VodBox) [inline UI/window-projector.hpp:22]: By changing the constant for 24 scenes in this enum, anyone who has 24 scenes selected will find themselves on the 18 scenes layout when launching OBS after this change.

Unfortunately the settings (incorrectly) uses the saved value as an index on the combox when first loading, despite the items in the combobox using the enum values as data independently of index.

A solution here to both problems would be to keep 24 scenes as 4, add 18 scenes as 5, and then use either findData() or a switch case on the setCurrentIndex in window-basic-settings.cpp.

**Review by VodBox — CHANGES_REQUESTED** [review]

File: UI/window-projector.hpp (hunk 1 above)
Comment (XDelta) [reply UI/window-projector.hpp]: Ah gotcha, made the change back for the order of the entries there.

**Comment by XDelta** [comment]
Sorted out the combobox setting working off of the index from enum using findData() as requested. Now having 24 scene multiview selected should find that its option is now at index 5 instead of relying on the enum index number directly.

**Pull request merged**